src/main/java/com/telefonica/iot/cygnus/sinks/OrionSink.java

```diff
@@ -245,13 +229,74 @@ public Status process() throws EventDeliveryException {
                     } else {
                         LOGGER.warn(e.getMessage());
                     } // if else if
+                    
+                    LOGGER.info("Finishing transaction (" + MDC.get(Constants.HEADER_TRANSACTION_ID) + ")");
+                    txn.commit();
+                    txn.close();
+                    return Status.READY;
                 } // if else
-            } finally {
-                LOGGER.info("Finishing transaction (" + MDC.get(Constants.HEADER_TRANSACTION_ID) + ")");
-            } // try catch finally
+            } // try catch
         } else {
+            // get and process as many events as the batch size
+            int currentIndex;
+            
+            for (currentIndex = accumulator.getAccIndex(); currentIndex < batchSize; currentIndex++) {
```

Comment:
Not exactly. I check for the timeout each time I'm going to read an event. The `for` loop does not ensure an event is read, it is only an attempt. If there are no events within the channel, then the `process()` method is exited; later, a new `process()` invocation will be done by Flume core. I admit by exiting and re-runing the `process()` method the timeouts are not triggered at exact time intervals, but the final behaviour is very close to that.